### PR TITLE
Create blog for ChatGPT

### LINKED ISSUE #3367: Create dbt Documentation and Testing 10x faster with ChatGPT
### Contact Details

pedro@hellosage.com

### I have read the dbt Developer Blog contribution guidelines.

- [X] I have read the dbt Developer Blog contribution guidelines.

### Which of these best describes you?

- [X] I am a dbt Community member or partner contributing to the Developer Blog
- [ ] I work for dbt Labs and am creating this issue for a community or marketing approved piece.

### What is the topic of your post?

Whether you are creating your pipelines into dbt for the first time or just adding a new model once in a while, good documentation and testing should always be a priority for you and your team. But good documentation and testing is time-consuming and a hassle to write. This blog post provides a set of simple guidelines to leverage ChatGPT in order to automate most of the work in generating the YAML files with model descriptions and tests.

### Link to an initial outline.

I have shared my experience with using ChatGPT to generate docs and tests with my assigned dbt Solution Architect Luis Leon (luis.leon@dbtlabs.com) and he suggested that I write this blog post.
Happy to change the form and content in any way you see fit.

This doc has several page but it's mostly due to very large images and tables of results. Actual content is < 1000 words and easily "trimmable".
https://docs.google.com/document/d/1gdrLOyCxuTs7JvGVCzrOV9_JcsLrSp0FHivVTR22rTQ/edit?usp=sharing

### PR DESCRIPTION
**Pull request by runleonarun**
resolves  https://github.com/dbt-labs/docs.getdbt.com/issues/3367

## What are you changing in this pull request and why?

Adding blog post by @pbritosa-sage along with @jasnonaz 

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

**Comment by pbritosa-sage** [comment]
@runleonarun Is there any action item for me here?

**Comment by gwenwindflower** [comment]
@runleonarun i fixed some artifacts throwing off formatting in the YAML snippets, also I talked with @jasnonaz and we think because ChatGPT performs poorly in the 'bonus' section and displays incorrect code, it might add more confusion than value and so we edited that final piece out.

otherwise good to go.

A note for the medium term: this piece contains a lot of pictures of text to show the output from ChatGPT, which is a big accessibility concern. I think as LLMs become baked into everything we do we're going to need a style point for how we manage prompts and output from LLMs in an accessible way. Not a blocker for this piece but something for yall to consider when there's capacity.

**Comment by runleonarun** [comment]
@pbritosa-sage you are good! We will get this published once I give it a last look and @gwenwindflower approves it!

**Comment by pbritosa-sage** [comment]
@runleonarun I added some requests in here: https://github.com/dbt-labs/docs.getdbt.com/issues/3367#issuecomment-1638813530 can you please take a look before publishing?

File: website/blog/2023-07-17-GPT-and-dbt-test.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+By now, everyone knows the wonders of the GPT models for code generation and pair programming so this shouldn’t come as a surprise. But **ChatGPT really shines** at inferring the context of verbosely named fields from database table schemas. So in this post I am going to help you 10x your documentation and testing speed by using ChatGPT to do most of the leg work for you.
+
+<!--truncate-->
+
+As a one-person Analytics team at [Sage](http://www.hellosage.com/) I had to create our dbt pipelines from the ground up. This meant 30+ of facts and dimensions into a Staging Layer, plus all of the following layers of augmented models and Mart tables. After the fact, we are talking about 3500+ lines of YAML that I was NOT excited to get started on. Fortunately for me, this was February 2023 and ChatGPT had just come out and boy, was I glad to use it. After a good dose of “prompt engineering” I managed to get most of my documentation and tests written out, only needing a few extra tweaks.
```
Comment (runleonarun) [inline website/blog/2023-07-17-GPT-and-dbt-test.md]:
```suggestion
As a one-person Analytics team at [Sage](http://www.hellosage.com/) I had to create our dbt pipelines from the ground up. This meant 30+ tables of internal facts and dimensions + external data into a Staging Layer, plus all of the following layers of augmented models and Mart tables. After the fact, we are talking about 3500+ lines of YAML that I was NOT excited to get started on. Fortunately for me, this was February 2023 and ChatGPT had just come out and boy, was I glad to use it. After a good dose of “prompt engineering” I managed to get most of my documentation and tests written out, only needing a few extra tweaks.
```

**Review by gwenwindflower — APPROVED** [review]

File: website/blog/2023-07-17-GPT-and-dbt-test.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+I copied this definition from Snowflake, and I would recommend that you always include both the name and type of the fields to get better results. This is especially helpful if you want to make sure ChatGPT correctly distinguishes between text, number, boolean or array fields, which may not be clear from the name alone.
+
+## Start out with simple prompts
+
+We all know how ChatGPT can digest very complex prompts, but as this is a tool for you to get the best results for your use case, I would recommend starting with simple, but specific, prompts.
+
+Opening ChatGPT with GPT4, my first prompt is usually along these lines:
+
+![Alt text](../static/img/blog/2023-07-17-GPT-and-dbt-test/image1.png)
+And the output of this simple prompt is nothing short of amazing:
```
Comment (runleonarun) [inline website/blog/2023-07-17-GPT-and-dbt-test.md:60]:
```suggestion

And the output of this simple prompt is nothing short of amazing:
```

**Comment by pbritosa-sage** [comment]
@runleonarun sorry, one super simple last change to the text!

"Fortunately for me, this was February 2023 and ChatGPT had just come out and boy, was I glad to use it."
into
"Fortunately for me, this was February 2023 and ChatGPT had just come out. And boy, was I glad to have it."

**Comment by pbritosa-sage** [comment]
Also, my bio seems to have a typo:

image

File: website/blog/2023-07-17-GPT-and-dbt-test.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
+---
+title: "Create dbt Documentation and Tests 10x faster with ChatGPT"
+description: "You cangit s use ChatGPT to infer the context of verbosely named fields from database table schemas."
```
Comment (runleonarun) [inline website/blog/2023-07-17-GPT-and-dbt-test.md]:
```suggestion
description: "You can use ChatGPT to infer the context of verbosely named fields from database table schemas."
```

File: website/blog/2023-07-17-GPT-and-dbt-test.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+By now, everyone knows the wonders of the GPT models for code generation and pair programming so this shouldn’t come as a surprise. But **ChatGPT really shines** at inferring the context of verbosely named fields from database table schemas. So in this post I am going to help you 10x your documentation and testing speed by using ChatGPT to do most of the leg work for you.
+
+<!--truncate-->
+
+As a one-person Analytics team at [Sage](http://www.hellosage.com/) I had to create our dbt pipelines from the ground up. This meant 30+ tables of internal facts and dimensions + external data into a Staging Layer, plus all of the following layers of augmented models and Mart tables. After the fact, we are talking about 3500+ lines of YAML that I was NOT excited to get started on. Fortunately for me, this was February 2023 and ChatGPT had just come out and boy, was I glad to use it. After a good dose of “prompt engineering” I managed to get most of my documentation and tests written out, only needing a few extra tweaks.
```
Comment (runleonarun) [inline website/blog/2023-07-17-GPT-and-dbt-test.md]:
```suggestion
As a one-person Analytics team at [Sage](http://www.hellosage.com/) I had to create our dbt pipelines from the ground up. This meant 30+ tables of internal facts and dimensions + external data into a Staging Layer, plus all of the following layers of augmented models and Mart tables. After the fact, we are talking about 3500+ lines of YAML that I was NOT excited to get started on. Fortunately for me, this was February 2023 and ChatGPT had just come out. And boy, was I glad to have it. After a good dose of “prompt engineering” I managed to get most of my documentation and tests written out, only needing a few extra tweaks.
```

**Comment by runleonarun** [comment]
OK @pbritosa-sage I made those changes and updated the images to be smaller (but you can click them to see them close up). If everything looks OK, then we can publish!

**Comment by pbritosa-sage** [comment]
Looks great! Thank you!

**Comment by runleonarun** [comment]
Thanks so much for this blog post, @pbritosa-sage!!

**Comment by pbritosa-sage** [comment]
@runleonarun thank you and everyone on the team for you editing and publishing work! What's the final link that I can share? Do you usually share these posts on any platform?

**Comment by runleonarun** [comment]
Here's the final link: https://docs.getdbt.com/blog/create-dbt-documentation-10x-faster-with-chatgpt

I will share it in a July release note like [this one from June](https://docs.getdbt.com/docs/dbt-versions/release-notes/June-2023/product-docs-jun#newguides%EF%B8%8Fblog-posts-and-faqs)

**Comment by runleonarun** [comment]
@jasnonaz or @gwenwindflower where do you usually share these?

**Comment by pbritosa-sage** [comment]
That's awesome, thank you!

**Pull request merged**